pkg/cryptoutil/keystore_message.go

```diff
@@ -644,18 +652,19 @@ func (m *MessageKeystore) updateCurrentKey(ctx context.Context, groupPK, pk cryp
 }
 
 // NewMessageKeystore instantiate a new MessageKeystore
-func NewMessageKeystore(s datastore.Datastore) *MessageKeystore {
+func NewMessageKeystore(s datastore.Datastore, logger *zap.Logger) *MessageKeystore {
 	return &MessageKeystore{
 		preComputedKeysCount: 100,
 		store:                dssync.MutexWrap(s),
+		logger:               logger.Named("message-ks"),
```

Comment:
Shouldn't you test if the logger is nil? If it is, instantiate a new one (or `NewNop()`), else set a new namespace (`logger = logger.Named("keystore")`).
If you don't do that, you will get a runtime error when you'll want to log.

Reply:
Yep